addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java

```diff
@@ -1119,6 +1120,9 @@ public String read() {
             } catch (IllegalStateException e) {
                 logger.debug("Scanner for {} threw IllegalStateException; scanner possibly closed", thing.getUID());
                 nextToken = null;
+            } catch (BufferOverflowException e) {
```

Comment:
This is a bug it should not be caught

Reply:
Do you have another suggestion on what to do about it, since it occurs inside Scanner?
```
2018-01-18 07:12:24.896 [WARN ] [handler.BigAssFanHandler$FanListener] - FanListener for bigassfan:fan:20F85EF18A40 had unhandled exception: null
java.nio.BufferOverflowException: null
        at java.nio.HeapCharBuffer.put(HeapCharBuffer.java:189) [?:?]
        at java.io.Reader.read(Reader.java:102) [?:?]
        at java.util.Scanner.readInput(Scanner.java:804) [?:?]
        at java.util.Scanner.next(Scanner.java:1369) [?:?]
        at org.openhab.binding.bigassfan.handler.BigAssFanHandler$ConnectionManager.read(BigAssFanHandler.java:1113) [242:org.openhab.binding.bigassfan:2.3.0.201801181206]
        at org.openhab.binding.bigassfan.handler.BigAssFanHandler$FanListener.readMessage(BigAssFanHandler.java:665) [242:org.openhab.binding.bigassfan:2.3.0.201801181206]
        at org.openhab.binding.bigassfan.handler.BigAssFanHandler$FanListener.waitForMessage(BigAssFanHandler.java:660) [242:org.openhab.binding.bigassfan:2.3.0.201801181206]
        at org.openhab.binding.bigassfan.handler.BigAssFanHandler$FanListener.listener(BigAssFanHandler.java:639) [242:org.openhab.binding.bigassfan:2.3.0.201801181206]
        at org.openhab.binding.bigassfan.handler.BigAssFanHandler$FanListener.access$0(BigAssFanHandler.java:633) [242:org.openhab.binding.bigassfan:2.3.0.201801181206]
        at org.openhab.binding.bigassfan.handler.BigAssFanHandler$FanListener$1.run(BigAssFanHandler.java:573) [242:org.openhab.binding.bigassfan:2.3.0.201801181206]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
```

Reply:
What is your intention, can you give any idea of the input and how big it is.

Or is that problem and is it sometimes an unlimited stream?

Reply:
Sure. The messages are very small. Here are some examples. All messages terminate with a right paren.  

I've read that the buffer inside Scanner is 1K. Even though that is pretty small, I've never seen a message that even got close to that size (other than a corrupt or poorly formatted message).

My concern with not handling it where it's thrown, is that it will be caught here and the listener will die.
https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java#L574

Of course, the other solution would be to rewrite the code to not use Scanner, or leave it the way it is.
```
(Master Bedroom Control;DEVICE;INDICATORS;ON)
(Master Bedroom Fan;FAN;BOOKENDS;1;7)
(Porch Control;SNSROCC;STATUS;UNOCCUPIED)
(Master Bedroom Fan;FAN;TIMER;CURR;0)
(Master Bedroom Control;DEVICE;SERVER;PRODUCTION)
(Master Bedroom Fan;FAN;TIMER;MAX;7)
(Master Bedroom Fan;FAN;TIMER;MIN;1)
(Porch Control;ERRORLOG;ENTRIES;MAX;10)
(Master Bedroom Fan;FAN;WHOOSH;STATUS;OFF)
(Master Bedroom Fan;FW;NAME;FW000007)
(Master Bedroom Control;ERRORLOG;ENTRIES;NUM;10)
(Master Bedroom Fan;FW;FW000007;2.4.1)
(Porch Control;FW;NAME;FW000004)
(Master Bedroom Fan;GROUP;LIST;Master Bedroom)
(Master Bedroom Fan;GROUP;ROOM;TYPE;2)
(Master Bedroom Control;SNSROCC;STATUS;UNOCCUPIED)
(Porch Control;FW;FW000004;2.4.1)
(Master Bedroom Fan;LEARN;MINSPEED;0)
```

Reply:
Let's do it.